fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -970,37 +975,35 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 		void init() override {}
 
 		struct OpenAction : TypedAction<Writer, OpenAction> {
-			std::shared_ptr<SharedRocksDBState> sharedState;
 			std::string path;
 			ThreadReturnPromise<Void> done;
 			Optional<Future<Void>>& metrics;
 			const FlowLock* readLock;
 			const FlowLock* fetchLock;
 			std::shared_ptr<RocksDBErrorListener> errorListener;
 			Counters& counters;
-			OpenAction(std::shared_ptr<SharedRocksDBState> sharedState,
-			           std::string path,
+			OpenAction(std::string path,
 			           Optional<Future<Void>>& metrics,
 			           const FlowLock* readLock,
 			           const FlowLock* fetchLock,
 			           std::shared_ptr<RocksDBErrorListener> errorListener,
 			           Counters& counters)
-			  : sharedState(sharedState), path(std::move(path)), metrics(metrics), readLock(readLock),
-			    fetchLock(fetchLock), errorListener(errorListener), counters(counters) {}
+			  : path(std::move(path)), metrics(metrics), readLock(readLock), fetchLock(fetchLock),
+			    errorListener(errorListener), counters(counters) {}
 
 			double getTimeEstimate() const override { return SERVER_KNOBS->COMMIT_TIME_ESTIMATE; }
 		};
 		void action(OpenAction& a) {
 			ASSERT(cf == nullptr);
 
 			std::vector<std::string> columnFamilies;
-			rocksdb::Options options = getOptions();
+			rocksdb::DBOptions options = sharedState->getDbOptions();
```

Comment:
Doesn't it make one more copy here?

Reply:
The copy is intentional for DBOptions and ColumnFamilyOptions, since they are not called frequently as ReadOptions, and a copy gives the opportunity to modify the options at the call site.